tap_workday/streams/helpers.py

```diff
@@ -0,0 +1,247 @@
+
+    def call_with_request_criteria(page, updated_since):
+        criteria = {"Page": page}
+        if updated_since:
+            criteria["Updated_Since"] = updated_since
+        return client.call(operation_name, Request_Criteria=criteria)
+
+    def call_with_page_arg(page, updated_since):
+        # updated_since is ignored here as not all APIs support it in this form
+        return client.call(operation_name, page=page)
+
+    def call_without_pagination():
+        return client.call(operation_name)
+
+    strategies = [
+        call_with_response_filter,
+        call_with_request_criteria,
+        call_with_page_arg,
+        lambda page, updated_since: call_without_pagination(),
+    ]
+
+    while page <= total_pages:
+        for strategy in strategies:
+            try:
+                response = strategy(page, updated_since)
+                break
+            except TypeError:
+                continue
+        else:
+            # If all strategies fail, raise the last error
+            raise RuntimeError(f"All pagination strategies failed for {operation_name}")
+
+        # Build filter dicts
+        response_filter = {"Page": page}
+        if updated_since:
+            # Many Workday APIs use 'Updated_Since' or similar in Request_Criteria or Response_Filter
+            response_filter["Updated_Since"] = updated_since
+
+        try:
+            # Try passing filter in Response_Filter
+            response = client.call(operation_name, Response_Filter=response_filter)
+        except TypeError:
+            # Fallback: try passing filter in Request_Criteria
+            try:
+                criteria = {"Page": page}
+                if updated_since:
+                    criteria["Updated_Since"] = updated_since
+                response = client.call(operation_name, Request_Criteria=criteria)
+            except TypeError:
+                # Fallback: try passing page only
+                try:
+                    response = client.call(operation_name, page=page)
+                except TypeError:
+                    # As a last resort, call with no pagination (single page)
+                    response = client.call(operation_name)
+
+        serialized = serialize_object(response)
+        records = safe_get_records(serialized, data_key)
+        all_records.extend(records)
+
+        # Extract pagination info
+        results = serialized.get("Response_Results", {})
+        # Defensive: handle missing or None values
+        total_pages_val = results.get("Total_Pages")
+        page_val = results.get("Page")
+        try:
+            total_pages = int(total_pages_val) if total_pages_val is not None else 1
+        except (ValueError, TypeError):
+            total_pages = 1
+        try:
+            current_page = int(page_val) if page_val is not None else page
+        except (ValueError, TypeError):
+            current_page = page
+        page = current_page + 1
```

Comment:
this can also be a new method altogether, thus reducing the complexity of this function.